### DISCUSSION TITLE
crit unable to find 'v1.53.3' - use 'latest'

### DISCUSSION THREAD

**Discussion by vyskvy** [golangci/golangci-lint#3965] [Q&A] [answered]
### Description of the problem

Im trying to install

golangci-lint on local linux env and i was following the documentation from the offical website

`curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -d -b $(go env GOPATH)/bin v1.53.3
`

and i got this:

```
golangci/golangci-lint info checking GitHub for tag 'v1.53.3'
golangci/golangci-lint debug http_download https://github.com/golangci/golangci-lint/releases/v1.53.3
golangci/golangci-lint crit unable to find 'v1.53.3' - use 'latest' or see https://github.com/golangci/golangci-lint/releases for details

```

any ideas?

**Comment by ldez** [comment]
Hello,

I copy paste your command and I don't reproduce your problem:

```console
$ docker run --rm -it golang:1.20-alpine sh
/go # apk add -q git curl
/go # curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -d -b $(go env GOPATH)/bin v1.53.3 
golangci/golangci-lint info checking GitHub for tag 'v1.53.3'
golangci/golangci-lint debug http_download https://github.com/golangci/golangci-lint/releases/v1.53.3
golangci/golangci-lint info found version: 1.53.3 for v1.53.3/linux/amd64
golangci/golangci-lint debug downloading files into /tmp/tmp.GFpGoN
golangci/golangci-lint debug http_download https://github.com/golangci/golangci-lint/releases/download/v1.53.3/golangci-lint-1.53.3-linux-amd64.tar.gz
golangci/golangci-lint debug http_download https://github.com/golangci/golangci-lint/releases/download/v1.53.3/golangci-lint-1.53.3-checksums.txt
golangci/golangci-lint info installed /go/bin/golangci-lint
/go # 
```

This is probability because you have a problem accessing GitHub API.

Duplicate of:
- https://github.com/golangci/golangci-lint/discussions/3757
- https://github.com/golangci/golangci-lint/discussions/2983
- https://github.com/golangci/golangci-lint/discussions/2605

**Comment by vyskvy** [comment]
> This is probability because you have a problem accessing GitHub API

what does this means, how to have the access?

**Comment by ldez** [comment]
can you do that:
```bash
curl -H "Accept:application/json" https://github.com/golangci/golangci-lint/releases/v1.53.3
```

the expected response is:

```console
$ curl -H "Accept:application/json" https://github.com/golangci/golangci-lint/releases/v1.53.3
{"id":108711254,"tag_name":"v1.53.3","update_url":"/golangci/golangci-lint/releases/tag/v1.53.3","update_authenticity_token":"ftw4NcpPj49yhtFoOJyQn6uBaefYWMQ3AGo07ckMEf5CCP0paZ6yOIiIZymHdRhOkczvypuIVRU3TayXvfHL9w","delete_url":"/golangci/golangci-lint/releases/tag/v1.53.3","delete_authenticity_token":"lGgI1ge-HYpROVeeRlfoZWT1pVw9mIXhmRoMpHYmUhQJj9s_AiQE3wVJoBd51acx_ZkCyY9RHXsBFPVRxg129A","edit_url":"/golangci/golangci-lint/releases/edit/v1.53.3"}
```

**Comment by vyskvy** [comment]
i got this
```
{"id":108711254,"tag_name":"v1.53.3","update_url":"/golangci/golangci-lint/releases/tag/v1.53.3","update_authenticity_token":"8EcviBM4ldPIhSl-qaDA4fYW1CMx058k_DLn5z2E9JR8cDpUL6e88fGg9O4EKZh316s80KOHCqDdZsfOPHpxQQ","delete_url":"/golangci/golangci-lint/releases/tag/v1.53.3","delete_authenticity_token":"vmoa7RWU-HshilWHdYYoi6HHn8Z5wK_0EAXkwqpegL1oPFTcs3wjnuVzUccrZ8vwg5eI6So434wi31LCi4iAyQ","edit_url":"/golangci/golangci-lint/releases/edit/v1.53.3"}
```

**Reply by ldez** [reply]
so it should work.

The script just does that:
```console
$ curl -H "Accept:application/json" https://github.com/golangci/golangci-lint/releases/v1.53.3 2>&1 | tr -s '\n' ' ' | sed 's/.*"tag_name":"//' | sed 's/".*//'
v1.53.3
```

https://github.com/golangci/golangci-lint/blob/91afddc9f85faca85f1bfce68e14e069c1ea6f5f/install.sh#L117-L121

https://github.com/golangci/golangci-lint/blob/91afddc9f85faca85f1bfce68e14e069c1ea6f5f/install.sh#L324-L334

what is your OS?

**Comment by vyskvy** [comment]
linux ubuntu

**Reply by ldez** [reply]
I think the problem is just related to the GitHub API rate limit (too many calls to the GitHub API) => wait a few minutes and retry.

**Comment by vyskvy** [comment]
i dont think so, because im struggling with these 4 last hours

**Reply by ldez** [reply]
If you retry before the end of the rate limit the problem will always be the same. 

Can you try this command:
```bash
curl -s -H "Accept:application/json" https://github.com/golangci/golangci-lint/releases/v1.53.3 | tr -s '\n' ' ' | sed 's/.*"tag_name":"//' | sed 's/".*//'
```

**Comment by vyskvy** [comment]
returns this

v1.53.3

but i found way this to work, it seems i have installed this already jsut instead running

 golangci-lint run

i use

 $HOME/go/bin/golangci-lint run
 
 and now it will work

**Reply by ldez** [reply, marked as answer]
You have misled me: your problem is not the download but the run.

`$GOPATH/bin` should be added to your `PATH`.

This is not related to golangci-lint but to your Go installation.

**Comment by vyskvy** [comment]
I just found that but its weird that download gives message I sent you, however thanks a lot for support!
